eox_nelp/stats/metrics.py

```diff
@@ -0,0 +1,136 @@
+"""eox-nelp Metrics file.
+
+functions:
+    get_cached_courses: Return visible courses.
+    get_course_metrics: Return the metric for the given course_key.
+    get_learners_metric: Return number of learners, for the visible courses.
+    get_instructors_metric: Return number of instructors, for the visible courses.
+    get_courses_metrics: Return metrics for the visible courses.
+"""
+from django.conf import settings
+
+from eox_nelp.edxapp_wrapper.branding import get_visible_courses
+from eox_nelp.edxapp_wrapper.modulestore import modulestore
+from eox_nelp.edxapp_wrapper.site_configuration import configuration_helpers
+from eox_nelp.edxapp_wrapper.student import CourseAccessRole, CourseEnrollment
+from eox_nelp.stats.decorators import cache_method
+
+
+@cache_method
+def get_cached_courses(tenant):  # pylint: disable=unused-argument
+    """
+    Returns the visible courses. This method is just a wrapper of get_visible_courses that
+    allows to cache the result.
+
+    Args:
+        tenant<str>: String tenant identifier(site.domain)
+
+    Return:
+        list[<CourseOverview>]: List of CourseOverview records.
+    """
+    return get_visible_courses()
+
+
+@cache_method
+def get_course_metrics(course_key):
+    """
+    This allows to get the course stats metrics based on the course key.
+
+    Args:
+        course_key<opaque-key>: Course identifier.
+
+    Return:
+        <Dictionary>: Contains the course's metrics.
+    """
+    stats_settings = getattr(settings, "STATS_SETTINGS", {})
+    course = modulestore().get_course(course_key)
+    chapters = course.get_children()
+
+    sequentials = []
+
+    for chapter in chapters:
+        sequentials += chapter.get_children()
+
+    verticals = []
+
+    for sequential in sequentials:
+        verticals += sequential.get_children()
+
+    allowed_block_types = stats_settings.get("ALLOWED_VERTICAL_BLOCK_TYPES", [])
```

Comment:
Here I think is not vertical. Only block_types or xblocks, or component but I prefer only `xblock_types`. 

To analize a tenant I have the problem here that you must control the tenant xblocks installed, ok this is could be a general setting for sagar but this set that we have to managed(know) the installed xblocks. Something that is interesting, but what happen if a new release of openedx come with a new xblock. We have to add this one and add it to the setting?
Or this forces to query one time mongo and get all the xblock types and configure correctly the setting.🤔